lxd/daemon_storage.go

```diff
@@ -316,6 +316,30 @@ func daemonStorageMove(s *state.State, storageType string, target string) error
 		return err
 	}
 
+	// Validate pool and volume exist.
+	err = s.DB.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
```

Comment:
This will need moving higher up due to similar issues with the section starting on line 274 using unvalidated input.

Suggest doing this lookup earlier and then replacing all uses of sourcePool and sourceVolume with the vars loaded from the DB for pool and volume respectively.

Reply:
Can you also update `daemonStorageValidate` to use a similar approach, i.e only use the user input args to load the DB records, and all internal use for pool and volume name use the vars from the DB.

Reply:
Actually, suggest we split daemonStorageValidate into multiple functions, with one just dealing with validating and returning the DB info and then the same method can be used both during setting the values as well as moving them.